cmd/infra/aws/destroy.go

```diff
@@ -78,8 +81,17 @@ func NewDestroyCommand() *cobra.Command {
 }
 
 func (o *DestroyInfraOptions) Run(ctx context.Context) error {
+	if o.AwsInfraGracePeriod.Seconds() == 0 {
+		o.AwsInfraGracePeriod = 30 * time.Minute
+	}
+
+	destroyInfraCtx, destroyInfraCtxCancel := context.WithTimeout(ctx, o.AwsInfraGracePeriod)
+	defer destroyInfraCtxCancel()
+
+	o.Log.Info(fmt.Sprintf("waiting %d s", int(o.AwsInfraGracePeriod.Seconds())))
```

Comment:
Carefully with the log entries, we usually set most of the things on conditions to avoid the "too much verbosity" issue. Maybe make sense to explore to update the reason of `InfrastructureReady` condition saying something like `Deprovisioning... Grace Period remaining X` or something like that (IMHO). WDYT @muraee ?

Reply:
yeah makes sense. Will I remove it for now and if needs be I can look into reporting it in the condition?

Reply:
Logs and conditions are signal for different problem spaces. Conditions are contractual signal for consumers subject to API support policies. Logs are non contractual UX for humans.
External clients like the cli should no often manipulate status for resources owned by controllers. Also at the time of deleting the infra, all the CRs should be gone anyways.
Agree on wanting to avoid unnecessary logs, but as far as I cans see this line would only log once which is fine. I'd consider even adding a line with the remaining timeout along the existing o.Log.Info("WARNING: error during destroy, will retry", "error", err.Error()).

Reply:
I changed the log to be more descriptive so it now looks like ` Infra destruction timeout set to 5 s`. I think when someone sets a timeout and the timeout duration set is reported back to him should be enough don't think there's a need for a countdown.

Reply:
What happens if the timeout is met? I'd expect we return an err and communicate deletion is moving ahead without infra deletion succeeding.

Reply:
Atm it exits out with an error and doesn’t proceed with the destruction.  Don’t think implementing what you explained above would be too much work however I think it would lead to confusion as there would be resources left over after a successful destruction. I think it makes more sense to leave it as is but I’m open to discussion.